From 65ca93f20dbf3a9ec47c98af8b562af21852db4e Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <ra5478@freescale.com>
Date: Sun, 14 Oct 2012 07:40:10 -0500
Subject: ENGR00229470-1 MX6SL-Add support for debug UART to be sourced from
 24MHz.

If "debug_uart" is specified in the command line, uart will
be sourced from 24MHz XTAL. This is required for getting the
correct power measurements on MX6SL.
Certain analog power optimizations are done only if ALL PLLs
are bypassed on MX6SL. To verify this path, we need to ensure
that UART is not sourced from PLL3.

Signed-off-by: Ranjani Vaidyanathan <ra5478@freescale.com>
---
 arch/arm/plat-mxc/cpu.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/plat-mxc/cpu.c b/arch/arm/plat-mxc/cpu.c
index 073c237..eb163ab 100755
--- a/arch/arm/plat-mxc/cpu.c
+++ b/arch/arm/plat-mxc/cpu.c
@@ -61,6 +61,16 @@ static int __init jtag_wfi_setup(char *p)
 	return 0;
 }
 early_param("jtag", jtag_wfi_setup);
+
+
+static int __init setup_debug_uart(char *p)
+{
+	uart_at_24 = 1;
+	return 0;
+}
+
+early_param("debug_uart", setup_debug_uart);
+
 /**
  * early_console_setup - setup debugging console
  *
-- 
1.7.9.5

